scom/stackstate_checks/scom/scom.py

```diff
@@ -3,60 +3,29 @@
-    response = session.post('http://' + scom_ip + '/OperationsManager/authenticate',
-                            auth=HttpNtlmAuth(domain + '\\' + username, password), json=request_credentials)
-    return str(response.status_code)
-
-
-def hasChild(serviceid, session, scom_ip, domain, username, password):
-    response = (session.get('http://' + scom_ip + '/OperationsManager/data/objectInformation/' + serviceid,
-                            auth=HttpNtlmAuth(domain + '\\' + username, password))).json()
-    service_data_tree = response.get("relatedObjects", [])
-    if not service_data_tree:
-        return False
-    else:
-        return True
-
-
-def get_entity_health_state(session, scom_ip, domain, username, password, healthIconUrl):
-    if re.search('StatusOkComplete', healthIconUrl, flags=re.IGNORECASE) or re.search('healthy_ex', healthIconUrl,
-                                                                                      flags=re.IGNORECASE):
-        return 'Healthy'
-    elif re.search('StatusWarning', healthIconUrl, flags=re.IGNORECASE) or re.search('warning_ex', healthIconUrl,
-                                                                                     flags=re.IGNORECASE):
-        return 'Deviating'
-    elif re.search('StatusError', healthIconUrl, flags=re.IGNORECASE) or re.search('critical_ex', healthIconUrl,
-                                                                                   flags=re.IGNORECASE):
-        return 'Error'
-    elif re.search('StatusUnknownGreen', healthIconUrl, flags=re.IGNORECASE) or re.search('maintenance_ex',
-                                                                                          healthIconUrl,
-                                                                                          flags=re.IGNORECASE):
-        return 'Not Monitored'
-    else:
-        return 'Unkown'
-
+import subprocess
 
 class SCOM(AgentCheck):
-    INSTANCE_TYPE = 'scom'
 
+    INSTANCE_TYPE = 'scom'
+    requests_counter = 0
     def get_instance_key(self, instance):
         if 'hostip' not in instance:
             raise ConfigurationError('Missing url in topology instance configuration.')
 
         instance_url = instance['hostip']
         return TopologyInstance(self.INSTANCE_TYPE, instance_url)
 
-    def sendAlerts(self, session, scom_ip, domain, username, password, component_id):
+    def init_session(self,session, auth_method, domain, username, password, scom_ip):
+        request_credentials = base64.b64encode(auth_method + ':' + domain + '\\' + username + ':' + password)
+        response = session.post('http://' + scom_ip + '/OperationsManager/authenticate',auth=HttpNtlmAuth(domain + '\\' + username, password), json=request_credentials)
+        return str(response.status_code)
+
+    def send_alerts(self, session,component_id, domain, username, password, scom_ip):
+        if self.requests_counter > self.requests_threshold:
```

Comment:
Can we log when the `requests_threshold` has been reached

Reply:
i log it at the end because if i add it after every threshold check, then it's going to be logged multiple times
self.log.info("maximum number of requests reached! "+str(self.requests_counter))
should i add it after each threshold check ?